gnd/src/main/java/com/google/android/gnd/persistence/local/room/FieldData.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.gnd.persistence.local.room;
+
+import androidx.room.Embedded;
+import androidx.room.Relation;
+import java.util.List;
+
+/**
+ * Represents relationship among FieldEntity, MultipleChoiceEntity and OptionEntity.
+ *
+ * <p>Querying any of the below data class automatically loads the field annotated as @Relation.
```

Comment:
Not sure I follow what this means. Is there another way to implement this that doesn't require a class with public mutable members and lazy loading of foreign key relations (iiuc)?

Reply:
The comment is quite confusing, I'll try to simplify.

Our goal is to fetch a Project object from the local database which is exactly similar to the one fetched from the remote db. That means, the project should contains all layers and the layers should then have the respective forms and so on.

When we do a fetch query for `ProjectEntity` in room, it just returns a list of projects and nothing else. So, to recursively fetch data for all the relations, I had to introduce another class. This was the recommended solution I could find in the official docs. https://developer.android.com/reference/androidx/room/Relation.html
Please suggest if you have other ideas.

>  Is there another way to implement this that doesn't require a class with public mutable members and lazy loading of foreign key relations

We later convert the list into a `ImmutableList` while converting the `*Data` objects into their respective model objects.

Any suggestions regarding naming this class or maybe packaging them separately so that it's less confusing?

Reply:
Alternatively, we could also add these relations to the Entity class directly so that these `*Data` classes can be removed. But that means all of the data is always loaded regardless of whether we need it or not. This way, we have an option to fetch from either a `*Data` class to load all data or fetch from an `*Entity` class to load only specific data.

Reply:
I tried to put the `@Relation` in the entity class too, but it doesn't work

Reply:
Thanks for doing the research on this one. I see the `Data` classes are wrapping both the Entity, and the `@Relation` Entities. Would it be possible to eliminate these classes and move the `@Relation` into the Entity class itself? That way the whole project would be loaded at once (which matches the current use), and we can eliminate the extra layer of indirection introduced by the `*Data` classes. Wdyt?

Reply:
> I tried to put the @Relation in the entity class too, but it doesn't work

Pointing to my previous comment.

My first attempt to put the relations int the Entity class didn't work. But I'll try to dig deeper and try to make it work. Removing the extra layer does sound tempting enough for this extra work.

Reply:
Saw your comment on the response later https://github.com/google/ground-android/pull/246#issuecomment-565855870

On it.